### PR TITLE
feat(backend): typescript

### PR DESCRIPTION
**Pull request by ulfgebhardt**
## 🍰 Pullrequest

This PR enabled the use of typescript.

### Issues

- fixes https://github.com/Ocelot-Social-Community/Ocelot-Social/issues/4416

### Todo

- [X] None

### CODE REVIEW

File: backend/src/schema/types/type/User.gql [hunk 1]
```diff
@@ -214,10 +214,21 @@ type Mutation {
...
   switchUserRole(role: UserRole!, id: ID!): User
 
   saveCategorySettings(activeCategories: [String]): Boolean
+
+  requestPasswordReset(email: String!): Boolean!
+  resetPassword(email: String!, nonce: String!, newPassword: String!): Boolean!
+  changePassword(oldPassword: String!, newPassword: String!): String!
+
+  # Get a JWT Token for the given Email and password
+  login(email: String!, password: String!): String!
```
Comment (Elweyn) [inline backend/src/schema/types/type/User.gql:233]: Got an error on my local machine:
```
{
  extensions: {code: "GRAPHQL_VALIDATION_FAILED"},
  message: "Field \"login\" of type \"User\" must have a selection of subfields. Did you mean \"login { ... }\"?",
}
```

**Review by Elweyn — DISMISSED** [review]
There is something with the login query on the webapp that don't work!
And `´yarn migrate:prod`` had also a problem but this could be since I am working on a local machine...
Everything else seem to work properly!

File: backend/src/schema/resolvers/user_management.spec.js [hunk 2]
```diff
@@ -270,7 +270,6 @@ describe('login', () => {
 
         it('responds with "Incorrect email address or password."', async () => {
           await respondsWith({
-            data: null,
```
Comment (Mogge) [inline backend/src/schema/resolvers/user_management.spec.js:273]: Does `data: undefined` work?

File: backend/package.json [hunk 3]
```diff
@@ -9,10 +9,10 @@
   "main": "src/index.js",
   "scripts": {
     "__migrate": "migrate --compiler 'js:@babel/register' --migrations-dir ./src/db/migrations",
-    "prod:migrate": "migrate --migrations-dir ./dist/db/migrations --store ./dist/db/migrate/store.js",
-    "start": "node dist/",
-    "build": "babel src/ -d dist/ --copy-files",
-    "dev": "nodemon --exec babel-node src/ -e js,gql",
+    "prod:migrate": "migrate --migrations-dir ./build/db/migrations --store ./build/db/migrate/store.js",
```
[the remark this answers is not in the document]
Comment (Mogge) [reply backend/package.json:12]: You need a command to run it. Like
```
yarn prod:migrate init
```
or
```
yarn prod:migrate up
```
I get another error:
```
Error: Cannot find module '/home/moriz/projects/Ocelot-Social/backend/build/db/migrate/store.js'
```

File: backend/package.json [hunk 4]
```diff
@@ -9,10 +9,10 @@
...
+    "start": "node build/",
+    "build": "tsc && mkdir -p build/middleware/helpers/email/templates/ && cp -r src/middleware/helpers/email/templates/*.html build/middleware/helpers/email/templates/ && mkdir -p build/middleware/helpers/email/templates/en/ && cp -r src/middleware/helpers/email/templates/en/*.html build/middleware/helpers/email/templates/en/ && mkdir -p build/middleware/helpers/email/templates/de/ && cp -r src/middleware/helpers/email/templates/de/*.html build/middleware/helpers/email/templates/de/ && mkdir -p build/schema/types/ && cp -r src/schema/types/*.gql build/schema/types/ && mkdir -p build/schema/types/enum/ && cp -r src/schema/types/enum/*.gql build/schema/types/enum/ && mkdir -p build/schema/types/scalar/ && cp -r src/schema/types/scalar/*.gql build/schema/types/scalar/ && mkdir -p build/schema/types/type/ && cp -r src/schema/types/type/*.gql build/schema/types/type/",
```
Comment (Mogge) [inline backend/package.json:14]: Can this be a script to improve readability?

File: backend/src/schema/types/type/User.gql (hunk 1 above)
Comment (Mogge) [reply backend/src/schema/types/type/User.gql:233]: I cannot reproduce this error

**Review by Mogge — CHANGES_REQUESTED** [review]

File: backend/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (ulfgebhardt) [reply backend/package.json:12]: image

You have to run `yarn build` before.

For me it seems to work as seen in the screenshot.

File: backend/package.json (hunk 4 above)
Comment (ulfgebhardt) [reply backend/package.json:14]: yes, can I do this in a separate ticket? I would do it next if thats acceptable

File: backend/src/schema/types/type/User.gql (hunk 1 above)
Comment (ulfgebhardt) [reply backend/src/schema/types/type/User.gql:233]: image

having it running locally (only neo4j in docker) in compiled version (`yarn build` and `yarn start` both in webapp and backend). Login works as expected.

**I cannot reproduce this error either**

**Review by Mogge — APPROVED** [review]

**Pull request merged**